### DISCUSSION TITLE
How to get a list of products/variants with prices from all the markets?

### DISCUSSION THREAD

**Discussion by wpsamurai** [Shopify/storefront-api-feedback#86] [Help]
As I read in the documentation about international pricing (https://shopify.dev/api/examples/international-pricing), to get a product price for a given market I have to pass country code in a context. Here is an example from the documentation:
```graphql
query allProducts @inContext(country: CA) {
  products(first: 1) {
    edges {
      node {
        variants(first: 1) {
          edges {
            node {
              priceV2 {
                amount
                currencyCode
              }
            }
          }
        }
      }
    }
  }
}

```

Is there a way to get a list of products/variants with prices from all the markets defined in a shop?

**Comment by johncraigcole** [comment]
With the Storefront API you'd only be able to do this a single currency at a time. You can get the available currencies for the shop with a localization query.

```
query getCountries{
   localization {
      availableCountries {
         currency {
           isoCode
           name
           symbol
         }
         isoCode
         name
         unitSystem
      }
   }
}
```
I know we're talking about the Storefront API here but I thought I'd also mention that the Admin API does support querying multiple [presentment prices](https://shopify.dev/api/admin-graphql/2022-01/objects/ProductVariant#connection-productvariant-presentmentprices) at the variant level in a single request. You wouldn't want to expose an Admin API token in the browser but depending on the use case perhaps the Admin API can supplement things.

**Comment by wpsamurai** [comment]
Hi @johncraigcole, thank you for your explanation.

As I learnt from [the documentation](https://shopify.dev/api/examples/multiple-currencies) `presentment currencies` are deprecated. Also aren't presentment currencies and international prices completely different features? When using presentment currency, prices are generated per CURRENCY. When using international pricing, prices are generated per COUNTRY and CURRENCY and you can have many countries with the same currency. Also international pricing allows you to manually adjust the price per country, which is not possible for presentment currencies.

**Reply by johncraigcole** [reply]
Hey @wpsamurai yeah your absolutely correct and I apologize for the confusion. 

The presentment currencies on the Admin API seem to pull international prices correctly when the country to currency is a 1:1 mapping. As you indicated, things breakdown when say USD is being used across multiple markets. I'll need to look into if Shopify plans to also mark the presentmentPrices connection on the variant as deprecated for the Admin API as well.

With the new international pricing and Shopify markets in mind, using the Admin API to query [priceLists](https://shopify.dev/api/admin-graphql/2022-01/queries/pricelists) would give you the accuracy you're looking for. Even though it's possible to pull all price lists at once, I prefer the approach of specifying a country matchRule and doing a country at a time.
